src/main/java/wolox/gmjavatraining/controllers/BookController.java

```diff
@@ -1,17 +1,75 @@
 package wolox.gmjavatraining.controllers;
 
-import org.springframework.stereotype.Controller;
-import org.springframework.ui.Model;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.DataAccessException;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.DeleteMapping;
 import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.server.ResponseStatusException;
+import wolox.gmjavatraining.exceptions.BookIdMismatchException;
+import wolox.gmjavatraining.exceptions.BookNotFoundException;
+import wolox.gmjavatraining.models.Book;
+import wolox.gmjavatraining.repositories.BookRepository;
 
-@Controller
+@RestController
+@RequestMapping("/api/books")
 public class BookController {
 
-    @GetMapping("/greeting")
-    public String greeting(@RequestParam(name="name", required=false, defaultValue="World") String name, Model model) {
-        model.addAttribute("name", name);
-        return "greeting";
+    @Autowired
+    private BookRepository bookRepository;
+
+    @GetMapping
+    public Iterable findAll() {
+        return bookRepository.findAll();
+    }
+
+    @PostMapping
+    @ResponseStatus(HttpStatus.CREATED)
+    public Book create(@RequestBody Book book) {
+        try {
+            return bookRepository.save(book);
+        } catch (DataAccessException ex) {
```

Comment:
For now this is more than Ok, but I believe in the link I shared with you, you could find how to write a Handler for this exception so that you don't even have to write the try catch structure here.